cmake/fmtlib.cmake

```diff
@@ -0,0 +1,31 @@
+include(ExternalProject)
+
+if (${CMAKE_SYSTEM_NAME} STREQUAL "Emscripten")
+	set(FMTLIB_CMAKE_COMMAND emcmake cmake)
+else()
+	set(FMTLIB_CMAKE_COMMAND ${CMAKE_COMMAND})
+endif()
+
+set(3rdparty_fmtlib_VERSION "7.1.3" CACHE STRING "fmtlib version")
+set(3rdparty_fmtlib_CHECKSUM "SHA256=5cae7072042b3043e12d53d50ef404bbb76949dad1de368d7f993a15c8c05ecc" CACHE STRING "fmtlib checksum")
+set(3rdparty_fmtlib_URL "https://github.com/fmtlib/fmt/archive/refs/tags/${3rdparty_fmtlib_VERSION}.tar.gz")
```

Comment:
Why the `3rdparty_` prefix? I see we are not using that convention with `jsoncpp.cmake` or `range v3.cmake`.

Reply:
well, yes, any name fits. The reasoning for that prefix is purely copy'n'paste of out of one of my own source repos where I am using fmtlib. I can change that.

Reply:
ok. I'd drop the prefix then.

Reply:
The other files do not use extra variables for this, just use it in place for ExternalProject/FetchContent.

Reply:
I wanted to suggest that too but I noticed that `3rdparty_fmtlib_VERSION` is used inside `3rdparty_fmtlib_URL` and I think it's actually a good thing to have a variable for version. Makes it easier to update and harder to forget to update the places where it is used.

Reply:
Sure, but makes the file I think needlessly complicated and none of the others use this pattern (rather the one I pushed).